tests_e2e/tests/agent_publish/agent_publish.py

```diff
@@ -68,6 +71,22 @@ def get_ignore_errors_before_timestamp(self) -> datetime:
         timestamp = self._ssh_client.run_command("agent_publish-get_agent_log_record_timestamp.py")
         return datetime.strptime(timestamp.strip(), u'%Y-%m-%d %H:%M:%S.%f')
 
+    def _wait_for_agent_to_complete_provisioning(self):
```

Comment:
I'd consider moving this somewhere in testss/lib, since it looks like a useful function that other tests could use